zetaclient/context/app.go

```diff
@@ -120,6 +125,27 @@ func (a *AppContext) GetKeygen() observertypes.Keygen {
 	}
 }
 
+func (a *AppContext) GetNodeAccounts() []observertypes.NodeAccount {
+	a.mu.RLock()
+	defer a.mu.RUnlock()
+
+	// deep copy node accounts
+	nodeAccounts := []observertypes.NodeAccount{}
```

Comment:
Can you elaborate on why a deep copy is required? 

Also, I think we need to implement `a.updateNodeAccounts()` that takes new accounts, sorts them, compares them with the current slice, and changes only if necessary (deterministically). In 99.99% of times it will be a `no-op`.